### PR TITLE
DBZ-3596 Removing some unused code

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.redhat.com/browse/DBZ-3596

@jpechane, @Naros, could one of you take a look? Thx!

### CODE REVIEW

**Comment by gunnarmorling** [comment]
The commits are fully independent. I'm 100% confident about the first one, 99% about the second one :) I don't think the "exported" flag should be needed any longer, given only `true` is ever passed in at the outer-most entry points after @jpechane's recent rework of all snapshot modes.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorConfig.java [hunk 1]
```diff
@@ -374,7 +374,7 @@ public boolean supportsTruncate() {
         DECODERBUFS("decoderbufs") {
             @Override
             public MessageDecoder messageDecoder(MessageDecoderConfig config) {
-                return new PgProtoMessageDecoder(config);
+                return new PgProtoMessageDecoder();
```
[the remark this answers is not in the document]
Comment (gunnarmorling) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorConfig.java:377]: > the config param should be dropped from the messageDecoder siganture method too.

It still is needed for the `PgOutputMessageDecoder`.

**Comment by gunnarmorling** [comment]
@jpechane, reworked as we discussed it.

**Comment by jpechane** [comment]
@gunnarmorling It lloks like it is broken, first there are exceptions like
```
2021-06-08T15:18:17.1523269Z 2021-06-08 15:18:17,149 ERROR  Postgres|test_server|streaming  Producer failure   [io.debezium.pipeline.ErrorHandler]
2021-06-08T15:18:17.1524663Z java.lang.NullPointerException: Connection does not provide type registry
2021-06-08T15:18:17.1525752Z 	at java.base/java.util.Objects.requireNonNull(Objects.java:246)
2021-06-08T15:18:17.1527861Z 	at io.debezium.connector.postgresql.connection.PostgresConnection.getTypeRegistry(PostgresConnection.java:578)
2021-06-08T15:18:17.1531175Z 	at io.debezium.connector.postgresql.connection.PostgresConnection.doReadTableColumn(PostgresConnection.java:548)
2021-06-08T15:18:17.1534605Z 	at io.debezium.connector.postgresql.connection.PostgresConnection.readColumnForDecoder(PostgresConnection.java:516)
2021-06-08T15:18:17.1539082Z 	at io.debezium.connector.postgresql.connection.pgoutput.PgOutputMessageDecoder.getTableColumnsFromDatabase(PgOutputMessageDecoder.java:334)
2021-06-08T15:18:17.1544120Z 	at io.debezium.connector.postgresql.connection.pgoutput.PgOutputMessageDecoder.handleRelationMessage(PgOutputMessageDecoder.java:272)
2021-06-08T15:18:17.1548729Z 	at io.debezium.connector.postgresql.connection.pgoutput.PgOutputMessageDecoder.processNotEmptyMessage(PgOutputMessageDecoder.java:176)
2021-06-08T15:18:17.1552838Z 	at io.debezium.connector.postgresql.connection.AbstractMessageDecoder.processMessage(AbstractMessageDecoder.java:33)
2021-06-08T15:18:17.1556282Z 	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.deserializeMessages(PostgresReplicationConnection.java:493)
2021-06-08T15:18:17.1559619Z 	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.readPending(PostgresReplicationConnection.java:485)
2021-06-08T15:18:17.1564662Z 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.processMessages(PostgresStreamingChangeEventSource.java:185)
2021-06-08T15:18:17.1568326Z 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.execute(PostgresStreamingChangeEventSource.java:147)
2021-06-08T15:18:17.1571298Z 	at io.debezium.pipeline.ChangeEventSourceCoordinator.streamEvents(ChangeEventSourceCoordinator.java:159)
2021-06-08T15:18:17.1573701Z 	at io.debezium.pipeline.ChangeEventSourceCoordinator.lambda$start$0(ChangeEventSourceCoordinator.java:122)
2021-06-08T15:18:17.1575407Z 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
2021-06-08T15:18:17.1586716Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2021-06-08T15:18:17.1588119Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-06-08T15:18:17.1590209Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-06-08T15:18:17.1600347Z 	at java.base/java.lang.Thread.run(Thread.java:829)
```

and later on it ends in an infinite loop

**Comment by gunnarmorling** [comment]
@jpechane, reworked a bit. I don't think the remaining failure is related; do you see this one locally?

**Comment by jpechane** [comment]
@gunnarmorling https://issues.redhat.com/browse/DBZ-3231 ?

**Comment by gunnarmorling** [comment]
Yeah, that seems to be the one.

>

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks!

**Pull request merged**